### PR TITLE
Package api

### PR DESCRIPTION
**Pull request by thetechnick**

### CODE REVIEW

File: apis/manifests/v1alpha1/packagemanifest_types.go [hunk 1]
```diff
@@ -2,31 +2,109 @@ package v1alpha1
...
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+)
+
+const (
+	// Package Phase annotation to assign objects to a phase.
+	PackagePhaseAnnotation = "package-operator.run/phase"
+)
+
+const (
+	// PackageLabel contains the name of the Package in the PackageManifest.
```
Comment (garrettthomaskth) [inline apis/manifests/v1alpha1/packagemanifest_types.go:16]:
```suggestion
	// PackageLabel contains the name of the Package from the PackageManifest.
```

File: apis/manifests/v1alpha1/packagemanifest_types.go [hunk 2]
```diff
@@ -2,31 +2,109 @@ package v1alpha1
...
+	AvailabilityProbes []corev1alpha1.ObjectSetProbe `json:"availabilityProbes"`
 }
 
 type PackageManifestPhase struct {
 	// Name of the reconcile phase. Must be unique within a PackageManifest
-	Name  string `json:"name" yaml:"name"`
-	Class string `json:"class,omitempty" yaml:"class,omitempty"`
+	Name string `json:"name"`
+	// If non empty, phase reconciliation is delegated to another controller.
+	// If set to the string "default" the built-in controller reconciling the object.
```
Comment (garrettthomaskth) [inline apis/manifests/v1alpha1/packagemanifest_types.go:58]:
```suggestion
	// If set to the string "default" the remote-phase-manager will reconcile the object in the same cluster.
	// If set to the string "hosted-cluster" the remote-phase-manager will reconcile the object in the given target cluster.
```

File: cmd/package-operator-manager/main.go [hunk 3]
```diff
@@ -77,13 +92,91 @@ func main() {
...
+}
+
+const packageFolderPath = "/package"
+
+func runLoader(scheme *runtime.Scheme, packageKey client.ObjectKey) error {
+	c, err := client.New(ctrl.GetConfigOrDie(), client.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		return fmt.Errorf("creating target cluster client: %w", err)
```
Comment (garrettthomaskth) [inline cmd/package-operator-manager/main.go:160]: Is this really the target cluster client? Like the hosted cluster?

File: apis/manifests/v1alpha1/packagemanifest_types.go (hunk 2 above)
Comment (thetechnick) [reply apis/manifests/v1alpha1/packagemanifest_types.go:58]: Nope, it's not part of the remote-phase-manager, it's embedded into the package-operator-manager:
https://github.com/package-operator/package-operator/blob/main/cmd/package-operator-manager/main.go#L177-L192

File: cmd/package-operator-manager/main.go (hunk 3 above)
Comment (thetechnick) [reply cmd/package-operator-manager/main.go]: My bad, copy/pasta issue :)
Should be fixed now!

File: internal/controllers/packages/adapter_package.go [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+func (a *GenericClusterPackage) GetSpecHash() string {
+	return utils.ComputeHash(a.Spec, nil)
+}
+
+func (a *GenericClusterPackage) setStatusPhase(phase corev1alpha1.PackageStatusPhase) {
+	a.Status.Phase = phase
+}
+
+func updatePackagePhase(pkg genericPackage) {
```
Comment (garrettthomaskth) [inline internal/controllers/packages/adapter_package.go:110]:
```suggestion
func updatePackagePhaseStatus(pkg genericPackage) {
```

**Pull request merged**